File: Add Code Here/PYTHON/potentiation.py
def pot(a, b):
    if b == 0 and a != 0:
        return 1
    if b >= 1:
        return a * pot(a, b - 1)
    if b <= -1:
        return 1.0 / a * pot(a, b + 1)
    return 0

def main():
    a, b = map(int, input().split())
    total = 0

    if a == 0 and b <= 0:
        print("undefined")
    else:
        total = pot(a, b)
        print(total)

if __name__ == "__main__":
    main()

File: Add Code Here/Java/potentiation.java
import java.util.Scanner;

public class Main {
    public static float pot(int a, int b) {
        if (b == 0 && a != 0)
            return 1;
        if (b >= 1)
            return a * pot(a, b - 1);
        if (b <= -1)
            return 1.0f / a * pot(a, b + 1);
        return 0;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int a, b;
        float total;
        a = scanner.nextInt();
        b = scanner.nextInt();

        if (a == 0 && b <= 0) {
            System.out.println("undefined");
        } else {
            total = pot(a, b);
            System.out.println(total);
        }
    }
}
